cli/azd/pkg/commands/pipeline/azdo_provider_test.go

```diff
@@ -0,0 +1,213 @@
+	t.Run("accepts a PAT via system environment variables", func(t *testing.T) {
+		// arrange
+		testPat := "12345"
+		provider := getEmptyAzdoScmProviderTestHarness()
+		os.Setenv(azdo.AzDoEnvironmentOrgName, "testOrg")
+		os.Setenv(azdo.AzDoPatName, testPat)
+		testConsole := &circularConsole{}
+		ctx := context.Background()
+
+		// act
+		e := provider.preConfigureCheck(ctx, testConsole)
+
+		// assert
+		require.NoError(t, e)
+
+		//cleanup
+		os.Unsetenv(azdo.AzDoPatName)
+	})
+
+	t.Run("returns an error if no pat is provided", func(t *testing.T) {
+		// arrange
+		os.Unsetenv(azdo.AzDoPatName)
+		os.Setenv(azdo.AzDoEnvironmentOrgName, "testOrg")
+		provider := getEmptyAzdoScmProviderTestHarness()
+		testConsole := &configurablePromptConsole{}
+		testPat := "testPAT12345"
+		testConsole.promptResponse = testPat
+		ctx := context.Background()
+
+		// act
+		e := provider.preConfigureCheck(ctx, testConsole)
+
+		// assert
+		require.Nil(t, e)
+		require.EqualValues(t, provider.Env.Values[azdo.AzDoPatName], testPat)
+	})
+
+}
+
+func Test_saveEnvironmentConfig(t *testing.T) {
+	tempDir := t.TempDir()
+
+	t.Run("saves to environment file", func(t *testing.T) {
+		// arrange
+		key := "test"
+		value := "12345"
+		provider := getEmptyAzdoScmProviderTestHarness()
+		envPath := path.Join(tempDir, ".test.env")
+		provider.Env = environment.EmptyWithFile(envPath)
+		// act
+		e := provider.saveEnvironmentConfig(key, value)
+		// assert
+		writtenEnv, err := environment.FromFile(envPath)
+		require.NoError(t, err)
+
+		require.EqualValues(t, writtenEnv.Values[key], value)
+		require.NoError(t, e)
+	})
+
+}
+func getEmptyAzdoScmProviderTestHarness() *AzdoScmProvider {
+	return &AzdoScmProvider{
+		Env: &environment.Environment{
+			Values: map[string]string{},
+		},
+	}
+}
+
+func getAzdoScmProviderTestHarness() *AzdoScmProvider {
+	return &AzdoScmProvider{
+		Env: &environment.Environment{
+			Values: map[string]string{
+				azdo.AzDoEnvironmentOrgName:       "fake_org",
+				azdo.AzDoEnvironmentProjectName:   "project1",
+				azdo.AzDoEnvironmentProjectIdName: "12345",
+				azdo.AzDoEnvironmentRepoName:      "repo1",
+				azdo.AzDoEnvironmentRepoIdName:    "9876",
+				azdo.AzDoEnvironmentRepoWebUrl:    "https://repo",
+			},
+		},
+	}
+}
+
+// For tests where the console won't matter at all
+type configurablePromptConsole struct {
+	promptResponse string
+}
+
+func (console *configurablePromptConsole) Message(ctx context.Context, message string) {}
+func (console *configurablePromptConsole) Prompt(ctx context.Context, options input.ConsoleOptions) (string, error) {
+	return console.promptResponse, nil
+}
+func (console *configurablePromptConsole) Select(ctx context.Context, options input.ConsoleOptions) (int, error) {
+	return 0, nil
+}
+func (console *configurablePromptConsole) Confirm(ctx context.Context, options input.ConsoleOptions) (bool, error) {
+	return false, nil
+}
+func (console *configurablePromptConsole) SetWriter(writer io.Writer) {}
+
```

Comment:
We already have a `MockConsole` implementation that I would prefer you use.  See it in some of the Bicep provision provider tests.

Reply:
Would that create a dependency from one test package to another?

Reply:
It's package in a test helper package. There's no official guidance on where test helper packages should live, either as `pkg` or `test`. We currently have it as a package under `test`.

Reply:
Updated to use MockConsole